specs/interop/predeploys.md

```diff
@@ -237,34 +240,60 @@ getters.
-|-------------------|--------------------------|-------------------------|---------|
-| {0x760ee04d}      | bytes4                   | 0-3                     | n/a     |
-| baseFeeScalar     | uint32                   | 4-7                     | 1       |
-| blobBaseFeeScalar | uint32                   | 8-11                    |         |
-| sequenceNumber    | uint64                   | 12-19                   |         |
-| l1BlockTimestamp  | uint64                   | 20-27                   |         |
-| l1BlockNumber     | uint64                   | 28-35                   |         |
-| basefee           | uint256                  | 36-67                   | 2       |
-| blobBaseFee       | uint256                  | 68-99                   | 3       |
-| l1BlockHash       | bytes32                  | 100-131                 | 4       |
-| batcherHash       | bytes32                  | 132-163                 | 5       |
-| interopSetSize    | uint8                    | 164-165                 | 6       |
-| chainIds          | uint256\[interopSetSize] | 165-(32*interopSetSize) | 6+      |
+### Static Configuration
+
+The `L1Block` contract MUST include method `setConfig(ConfigType, bytes)` for setting static values that are used by the
+system. This function serves to reduce the size of the L1 Attributes transaction, as well as to reduce the need to add
+specific one off functions.
+
+The `ConfigType` enum is defined as follows:
+
+```solidity
+enum ConfigType {
+    GAS_PAYING_TOKEN,
+    ADD_DEPENDENCY,
+    REMOVE_DEPENDENCY
+}
+```
+
+The second argument to `setConfig` is a `bytes` value that is ABI encoded with the necessary values for the `ConfigType`.
+
+| ConfigType          | Value                                       |
+|---------------------|---------------------------------------------|
+| `GAS_PAYING_TOKEN`  | `abi.encode(token, decimals, name, symbol)` |
+| `ADD_DEPENDENCY`    | `abi.encode(chainId)`                       |
+| `REMOVE_DEPENDENCY` | `abi.encode(chainId)`                       |
+
+where
+
+- `token` is the gas paying token's address (type `address`)
+
+- `decimals` is the gas paying token's decimals (type `uint8`)
+
+- `name` is the gas paying token's name (type `bytes32`)
+
+- `symbol` is the gas paying token's symbol (type `bytes32`)
+
+- `chainId` is the chain id intended to be added or removed from the dependency set
+
+Calls to `setConfig` MUST originate from `SystemConfig` and are forwarded to `L1Block` by `OptimismPortal`.
+
+### Dependency Set
+
+`L1Block` is updated to include the set of allowed chains. These chains are added and removed through `setConfig` calls
+with `ADD_DEPENDENCY` or `REMOVE_DEPENDENCY`, respectively.
+
+`L1Block` MUST provide a public getter to check if a particular chain is in the dependency set called
+`isInDependencySet(uint256)`. This function MUST return true when the chain's chain id is passed in as an argument.
```

Comment:
Can we also add another function `dependencySet()([]uint256)` as a way to fetch all of the dependencies?

Reply:
good idea!